gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/exception/DefaultPubSubMessageReceiverExceptionHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.exception;
+
+import io.micronaut.context.annotation.Primary;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+/**
+ * Handles any exception on beans of type {@link io.micronaut.gcp.pubsub.annotation.PubSubListener} that do not
+ * implement {@link PubSubMessageReceiverExceptionHandler} interface.
+ * Only logs the error. Messages will not be acked and will be redelivered for as long the subscription configuration
+ * is set for message retries.
+ *
+ * @author Vinicius Carvalho
+ * @since 2.0.0
+ */
+@Singleton
+@Primary
```

Comment:
Why is this annotated with `@Primary`?

Reply:
Honestly I was just following the example at https://github.com/micronaut-projects/micronaut-rabbitmq/blob/master/rabbitmq/src/main/java/io/micronaut/rabbitmq/exception/DefaultRabbitListenerExceptionHandler.java#L34. I noticed that some of my tests would fail as well without `@Primary` if I have multiple listeners that implement that interface and then `PubSubConsumerAdvice` would not be instantiated due multiple bean instances.

Reply:
The implementation only supports a single bean, so users should be replacing the default implementation with their own instead of just adding another bean. Primary is only when there _should_ be multiple beans and the annotation designates which bean should be injected when only a single instance is requested.